rules/visibility/ec2-missing-osquery.sql

```diff
@@ -0,0 +1,53 @@
+CREATE OR REPLACE VIEW rules.SERVER_MISSING_OSQUERY_AGENT_VIOLATION_QUERY COPY GRANTS
+  COMMENT='Violation rule for identifying visibility gaps where a server is missing the osquery agent
+  @id 2060a772e50e4a4598808cdb76d2a315
+  @tags servers, osquery, visibility gap'
+AS
+SELECT OBJECT_CONSTRUCT(
+         'cloud', 'aws',
+         'deployment', MAP.DEPLOYMENT,
+         'account', MAP.ACCOUNT_ALIAS,
+'fix_parameters', OBJECT_CONSTRUCT('ip_address', details:"PrivateIpAddress", 'account_name', MAP.account_alias, 'account_id', account_id, 'instance_id', details:"InstanceId", 'deployment', deployment)
+       ) AS environment
+     , instances AS object
+     , 'AWS EC2 instance expected osquery data missing' AS title
+     , CURRENT_TIMESTAMP() AS alert_time
+     , 'The affected server does not appear to be shipping osquery logs. Agent may be missing, broken, or unable to transmit.' AS description
+     , any_value(details) AS event_data
+     , 'SnowAlert' AS detector
+     , 'high' AS severity
+     , '2060a772e50e4a4598808cdb76d2a315' AS query_id
+     , 'owner-name' as owner
+     , OBJECT_CONSTRUCT('query_id', query_id, 'cloud', 'aws', 'account_id', accountid, 'instance_id', instances) AS identity
+FROM (
+  SELECT timestamp as snapshot_at
+       , instance:InstanceId::String as instances
+       , instance:Tags::String as tags
+       , instance:State:Name::String as state
+       , instance:"AccountId" as accountid
+       , instance:Platform::String as platform
+  , instance:"LaunchTime" as launch_time
+       , instance as details
+  FROM snowalert.base_data.ec2_instance_snapshots_t as snap
```

Comment:
The only thing I ran into was this `base_data.ec2_instance_snapshots_t` and `security.prod.osquery_v` seem to deviate from the current naming conventions.` I think some columns might be different because of this.